js/client-battle-tooltips.js

```diff
@@ -1205,12 +1229,56 @@ var BattleTooltips = (function () {
 		if (auraBoosted) {
 			if (auraBroken) {
 				basePower = Math.floor(basePower * 0.75);
-				basePowerComment = this.makePercentageChangeText(0.75, auraBoosted + ' + Aura Break');
+				basePowerComment += this.makePercentageChangeText(0.75, auraBoosted + ' + Aura Break');
 			} else {
 				basePower = Math.floor(basePower * 1.33);
-				basePowerComment = this.makePercentageChangeText(1.33, auraBoosted);
+				basePowerComment += this.makePercentageChangeText(1.33, auraBoosted);
+			}
+		}
+
+		// Field Effects
+		if (thereIsWeather) {
+			if ((this.battle.weather === 'sunnyday' || this.battle.weather === 'desolateland' && move.type === 'Fire') ||
+				(this.battle.weather === 'raindance' || this.battle.weather === 'primordialsea' && move.type === 'Water')) {
+				basePower = Math.floor(basePower * 1.5);
+				basePowerComment += this.makePercentageChangeText(1.5, 'weather');
+			}
+		}
+		var isGrounded = true;
+		var noItem = !pokemonData.item || this.battle.hasPseudoWeather('Magic Room') || pokemonData.volatiles && pokemonData.volatiles['embargo'];
+		if (this.Battle.hasPseudoWeather('Gravity')) {
+			isGrounded = true;
+		} else if (pokemonData.volatiles && pokemonData.volatiles['ingrain'] && this.battle.gen >= 4) {
+			isGrounded = true;
+		} else if (pokemonData.volatiles && pokemonData.volatiles['smackdown']) {
+			isGrounded = true;
+		} else if (!noItem && pokemonData.item === 'ironball') {
+			isGrounded = true;
+		} else if (ability === 'levitate') {
+			isGrounded = false;
+		} else if (pokemonData.volatiles && (pokemonData.volatiles['magnetrise'] || pokemonData.volatiles['telekinesis'])) {
+			isGrounded = false;
+		} else if (!noItem && pokemonData.item !== 'airballoon') {
+			isGrounded = false;
+		} else if (!(pokemonData.volatiles && pokemonData.volatiles['roost'])) {
+			// If a Fire/Flying type uses Burn Up and Roost, it becomes ???/Flying-type, but it's still grounded.
+			for (var i = 0; i < types.length; i++) {
+				if (types[i] === 'Flying') isGrounded = false;
```

Comment:
> still grounded

> `isGrounded = false;`

Reply:
That just explains why it checks for roost before it checks the typing; I literally copied this straight out of `isGrounded` in the server's sim code, complete with the comment.

Reply:
That said, I don't actually think it's been tested what happens if a flying/??? type that has roosted also has levitate or something else that lifts it up...